DietPi-Survey | Collected data adjustments, source-able survey file and further simplification

MichaIng opened the pull request:
**Status**: Ready for testing and review
- [x] Patch_file fix: dietpi-survey command argument + settings file + fix DietPi-Version (increase to target version before uploading)
- [x] Adjust collected data, survey version and count? Automation!
- [x] Make survey files source-able by survey-report script: `array[$UNIQUE_ID]=$value` or `((array[$value]++))` ?

**Reference**: https://github.com/Fourdee/DietPi/issues/1827

**Commit list/description**:
+ DietPi-Survey | Further simplification: Just two modes ($1=1 non-interactive, else interactive), opt choices = 0/1 instead of 0/2
+ DietPi-Survey | Silently fail in non-interactive mode
+ DietPi-Survey | Only show successful purge message in interactive mode
+ DietPi-Patch_file | Fix dietpi-survey command argument, settings file and DietPi-Version (increase to target version before uploading)
+ DietPi-Survey | Force interactive mode, if invalid opt-in/out setting is found
+ DietPi-Survey | Switch survey file to source-able format to speed up automated survey report web page generation
+ DietPi-Survey | Minor wording fix
+ DietPi-Survey | Added active network interface and DietPi-Automation info to survey file
+ DietPi-Survey | Patch: Whoopsie, values need to be written back to settings file

MichaIng committed:
v6.10

+ DietPi-Survey | Further simplification: Just two modes ($1=1 non-interactive, else interactive), opt choices = 0/1 instead of 0/2
+ DietPi-Survey | Silently fail in non-interactive mode
+ DietPi-Survey | Only show successful purge message in interactive mode

MichaIng committed:
v6.10

+ DietPi-Patch_file | Fix dietpi-survey command argument, settings file and DietPi-Version (increase to target version before uploading)

Fourdee commented on dietpi/dietpi-survey:
```diff
@@ -128,7 +126,7 @@ _EOF_
 DietPi-Survey v$SURVEY_VERSION
 -------------------------
-Upload Count   : $SURVEY_SENTCOUNT
+Upload Count   : $((SURVEY_SENTCOUNT+1))
```
This will be +1 than actual, at end of ```dietpi-update``` which also sends a survey.

MichaIng replied:
@Fourdee 
Currently (dietpi-update v6.9) does not call dietpi-survey yet. For this reason, new or pre 6.9 users would upload survey with sentcount=0, as we add +1 after upload, or do I miss something?

Fourdee commented on dietpi/patch_file:
```diff
@@ -835,6 +813,33 @@ _EOF_
 			#-------------------------------------------------------------------------------
+			#DietPi-Survey rework phase 2: https://github.com/Fourdee/DietPi/issues/1827#issuecomment-395970075, https://github.com/Fourdee/DietPi/pull/1884
+			local fp_temp='/DietPi/dietpi/.dietpi-survey'
+			if [[ -f $fp_temp ]]; then
```
File wont exist due to removal above?
https://github.com/Fourdee/DietPi/pull/1884/files#diff-80dee702d26909604441f6260ba5ba00R713

It may be more simple/efficient to simply remove all previous patches for survey, then prompt user again to setup with new system?

MichaIng replied:
@Fourdee 
Removal is and was only done for pre-6.9 updates, which now forces interactive mode.
If user updates from 6.9 to 6.10, settings file will be present.
I think we should not force user interaction again, if choice was already made. Also existing upload count would be lost.
The latter could be completely removed though, if we don't see too much use for it, but this is another question.

Fourdee commented on dietpi/dietpi-survey:
```diff
@@ -146,55 +144,42 @@ $(grep '=2' /DietPi/dietpi/.installed)
 $(<$UPLOAD_FILENAME)
 
-Would you like to opt in for DietPi-Survey?"
+Would you join DietPi-Survey?"
```
Minor language, some options:
- "Would you like to join the DietPi-Survey?"
- "Join the DietPi-Survey?"

MichaIng replied:
@Fourdee 
Ah yes that's true, crippled sentence currently, whoopsie 🤣. €: Fixed

MichaIng committed:
v6.10

+ DietPi-Survey | Force interactive mode, if invalid opt-in/out setting is found
+ DietPi-Survey | Switch survey file to source-able format to speed up automated survey report web page generation
+ DietPi-Survey | Minor wording fix

MichaIng committed:
v6.10

+ DietPi-Survey | Syntax

MichaIng commented:
@Fourdee 
Okay, I implemented now sourcing style how it should be fastest. But this is just an offer, as said. Creating the arrays with user specific index has other advantages and should be much faster then current scraping as well.

Software index to name translation is now DietPi version specific, so we need to copy and edit the translation array within survey-report according to changed software indexes: `aSOFTWARE_NAME6_10`

I will add handling of new file format to the report script, if you agree, before release.

MichaIng committed:
v6.10

+ DietPi-Survey | Patch: Whoopsie, values need to be written back to settings file

Fourdee commented:
@MichaIng 

> I will add handling of new file format to the report script, if you agree, before release.

Please 👍 

Merged for testing, great work 👍
